src/main/java/com/example/mresolver362/DependencyResolver.java

```diff
@@ -91,6 +88,7 @@ final class DependencyResolver {
 		try {
 			ServiceLocator serviceLocator = createServiceLocator();
 			DefaultRepositorySystemSession session = MavenRepositorySystemUtils.newSession();
+			session.setIgnoreArtifactDescriptorRepositories(true);
```

Comment:
This is not needed, I just tried to optimize.
with this set, you need to know ahead of time the repositories needed to resolve tree.
without this set (as in your original code), resolver may "wander" (and is wander) across MANY OTHER repositories that are badly set up.

See more here:
* https://maven.apache.org/resolver/remote-repository-filtering.html
* https://github.com/cstamas/rrf-demo